### ISSUE TITLE
More informations

### ISSUE THREAD

**Issue by atropo** [Afterster/videojs-java#1]
Please provide more informations on how to use this plugin: how to generate the jar file and where to place it. 
I'm having problems to make this work.

Thanks

**Comment by Afterster** [comment]
The jar file can be build simply using NetBeans, but using an IDE is optional. You can also directly download it from the latest release https://github.com/Afterster/videojs-java/releases/download/0.1.0/videojs-java-0.1.0.zip
You should place the jar side by side to your html page. This can be moved with `videojs.options.java.jar` javascript variable.

**Comment by moesingh** [comment]
to build the jar, you have to first download JMF from Sun via http://www.oracle.com/technetwork/java/javase/download-142937.html (get the zip file).  then you have to make a directory in the java/video-js directory called "lib" and put the jmf.jar and the mediaplayer.jar in there.  then you run the ant job via eclipse, intellij, etc. or just run ant in the java/video-js folder.  you need java jdk installed on your pc.  once done, you'll get a jar file int he folder video.js/dist

let me know if you need help or just need the jar file (i compiled with java 1.7)

**Comment by DonkeyInBeijing** [comment]
I am recently using this on my project,  I place the video-js.jar in the same folder with my html and get lots of error as below:
`Uncaught TypeError: Cannot read property 'extend' of undefined(anonymous function) @ videojs.java.js:1
index.html:41 Uncaught TypeError: Cannot set property 'jar' of undefined(anonymous function) @ index.html:41
2video.js:19296 VIDEOJS: ERROR: The "Java" tech is undefined. Skipped browser support check for that tech._logType @ video.js:19296137.log.error @ video.js:19247(anonymous function) @ video.js:11193selectSource @ video.js:11183sourceList_ @ video.js:11348src @ video.js:11291MediaLoader @ video.js:15244addChild @ video.js:3562handleAdd @ video.js:3724(anonymous function) @ video.js:3770initChildren @ video.js:3771Player @ video.js:9536videojs @ video.js:19811autoSetup @ video.js:12813
video.js:19296 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) No compatible source was found for this media. MediaError {code: 4, message: "No compatible source was found for this media."}_logType @ video.js:19296137.log.error @ video.js:19247error @ video.js:11662(anonymous function) @ video.js:11361ret @ video.js:19125
video.js:19296 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) No compatible source was found for this media. MediaError {code: 4, message: "No compatible source was found for this media."}`
Please help me out of this.
thank  you!

**Comment by Afterster** [comment]
Unfortunately I will not update the plugin anymore for now. This mean it may not work on VideoJS 5 in current state.

All my videojs-\* tech plugins was done in attempt/prevision to migrate [Ampache](https://github.com/ampache/ampache) from jPlayer to VideoJS but this wasn't done (yet) because of several technical limitations. I really miss time here and had to refocus myself to not fail all :smile:.

I can transfer project ownership to anyone willing to continue the work.

**Comment by DonkeyInBeijing** [comment]
@Afterster I did not focus on the back compatibility anymore. : ) anyway, Thank you.

**Comment by andy012** [comment]
@DonkeyInBeijing   I meet the same problem as you, 

`
videojs-java.js:1 Uncaught TypeError: Cannot read property 'extend' of undefined
MyStream.html:31 Uncaught TypeError: Cannot set property 'jar' of undefined
video.min.js:22VIDEOJS: ERROR: The "Java" tech is undefined. Skipped browser support check for that tech.e @ video.min.js:22
video.min.js:22VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) No compatible source was found for this media. ie @ video.min.js:22
.........
`
are you solve the problem?
thank you!

**Comment by salisbury-espinosa** [comment]
@andy012 This error occurs in the latest version videojs , according to the file https://github.com/Afterster/videojs-java/blob/master/package.json , you should use an older version

```
"peerDependencies": {
    "video.js": "^4.5"
  }
```

**Comment by wishma2013** [comment]
I use video.js 5.15.1 and videojs-java.js .I correct the first line code in videojs-java.js "videojs.MediaTechController.extend" to "videojs.getComponent("MediaTechController").extend",and   then it thow warns and error like below:
video.js:22914 VIDEOJS: WARN: Component.extend({}) has been deprecated,  use videojs.extend(Component, {}) instead 

video.js:22914 VIDEOJS: WARN: The Java component was added to the videojs object when it should be registered using videojs.registerComponent(name, component)

video.js:22914 VIDEOJS: WARN: The Java tech was added to the videojs object when it should be registered using videojs.registerTech(name, tech)

video.js:22914 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) No compatible source was found for this media. 

I download the videohs-java.jar on Internet.

I cannot find out the problem.

**Comment by wishma2013** [comment]
I'm trying to let videojs support MOV video form via videojs-java.js .but It likes to failed.I try many versions of video.js,throw error "No compatible source was found for this video." all the time.if video.js cannot support play MOV video at all?

**Comment by tarunsharma1** [comment]
Has anyone got this plugin to work? 
I have tried with older versions of video js but still no luck..I keep getting the error 'video.js:24 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) No compatible source was found for this video'

Has anyone solved this?

**Comment by nzhtrtlc** [comment]
@tarunsharma1 I have the same problem either.

**Comment by cassiewang01** [comment]
throw error "No compatible source was found for this video
@wishma2013   @tarunsharma1 @nzhtrtlc Has anyone solved this?

**Comment by nzhtrtlc** [comment]
@cassiewang01  Nope, i switched standart html5 video.

**Comment by aefox** [comment]
@Afterster I would be interested to continue work on this project. is the offer still available?

**Comment by Afterster** [comment]
Yes it is, just send a PR for starting and then I will make you project manager. Otherwise you can just fork it and keep it alive at your fork. Your decision here, let me know :smile:

**Comment by wishma2013** [comment]
I am very pleased to have this opportunity to be manager of project and very glad to work with you.
